### PR TITLE
Initial stage of channel based version.

### PR DESCRIPTION
**Pull request by w-vi**
It spawns goroutines for log processing and then sending to statsd, each
having a buffer channel as input. Buffer size is a constat to change it
you need to rebuild it at the moment. Default size is 500.
- Very rough, although with some test.
- Not really optimized yet.
- Needs more test for edge cases.
- Godep save updated

### CODE REVIEW

**Comment by w-vi** [comment]
@abtris If tests on 1.4 are ok we can go forward I think and switch to 1.5 only when usptream datadog will merge the 1.5 fix.

File: server.go [hunk 1]
```diff
@@ -1,27 +1,167 @@
...
+//STATSD_URL=..  Optional. Default: statsd://localhost:8125
+//DATADOG_DRAIN_DEBUG=         Optional. If DEBUG is set, a lot of stuff w
+func loadServerCtx() *ServerCtx {
+
+	s := &ServerCtx{"8080",
+		nil,
+		make(map[string]string),
+		make(map[string][]string),
+		make(map[string]string),
+		"statsd://localhost:8125",
```
Comment (abtris) [inline server.go]: this can be set by `STATSD_URL`

**Comment by abtris** [comment]
:+1:
